### PR TITLE
ci: use parallel instead of MADSIM_TEST_NUM for scale test

### PR DESCRIPTION
**Pull request by xxchan**
I hereby agree to the terms of the [RisingWave Labs, Inc. Contributor License Agreement](https://gist.github.com/TennyZhuang/f00be7f16996ea48effb049aa7be4d66#file-rw_cla).

## What's changed and what's your intention?
Try to mitigate #7561

## Documentation

- [x] My PR **DOES NOT** contain user-facing changes.

<details><summary>Click here for Documentation</summary>

### Types of user-facing changes

Please keep the types that apply to your changes, and remove the others.

- Installation and deployment
- Connector (sources & sinks)
- SQL commands, functions, and operators
- RisingWave cluster configuration changes
- Other (please specify in the release note below)

### Release note

</details>

### CODE REVIEW

**Comment by xxchan** [comment]
Recent fail of main-cron is scaling test, not recovery test...

**Comment by BugenZhao** [comment]
> Then why does it run out of memory? (or does it do so?)

Possible if there's some process-level leak in our system. 🥵

**Comment by fuyufjh** [comment]
Perhaps we need to reproduce it on EC2 machine first...

File: ci/scripts/deterministic-e2e-test.sh [hunk 1]
```diff
@@ -21,23 +21,40 @@ export LOGDIR=.risingwave/log
 
 mkdir -p $LOGDIR
 
+function run_parallel_jobs {
+  # Limit the number of parallel jobs to avoid OOM.
+  # See https://github.com/risingwavelabs/risingwave/issues/7561
+  MAX_JOBS=32
```
Comment (wangrunji0408) [inline ci/scripts/deterministic-e2e-test.sh:27]: You can simply use `parallel -j32` to achieve the same effect.

**Comment by wangrunji0408** [comment]
> Recent fail of main-cron is scaling test, not recovery test... It uses MADSIM_TEST_NUM, instead of parallel.
> Wait, so we are runing scaling test and unit test in serial? @wangrunji0408 @BugenZhao
> Then why does it run out of memory? (or does it do so?)

Yes, using `MADSIM_TEST_NUM` runs in serial, and in the same process. So it's vulnerable to memory leaks.
Maybe we can migrate it to `parallel` and see if the OOM will be fixed.

**Comment by xxchan** [comment]
It looks good after `parallel`...

image

**Comment by xxchan** [comment]
```
MADSIM_TEST_NUM=100 NEXTEST_PROFILE=ci-scaling r+ sscale-test nexmark_chaos_q9_single --nocapture
```
The memory keeps increasing after each run 🥵
```
seed: 1677874297170683775
used memory : 11206 MB
used swap   : 0 MB
seed: 1677874297170683776
used memory : 11521 MB
used swap   : 0 MB
seed: 1677874297170683777
used memory : 11989 MB
used swap   : 0 MB
seed: 1677874297170683778
used memory : 12292 MB
used swap   : 0 MB
        SLOW [>400.000s] risingwave_simulation::it nexmark_chaos::nexmark_chaos_q9_single
seed: 1677874297170683779
used memory : 12587 MB
used swap   : 0 MB
seed: 1677874297170683780
used memory : 12912 MB
used swap   : 0 MB
seed: 1677874297170683781
used memory : 13227 MB
used swap   : 0 MB
seed: 1677874297170683782
used memory : 13531 MB
used swap   : 0 MB
seed: 1677874297170683783
used memory : 13855 MB
used swap   : 0 MB
        SLOW [>800.000s] risingwave_simulation::it nexmark_chaos::nexmark_chaos_q9_single
seed: 1677874297170683784
used memory : 14179 MB
used swap   : 0 MB
seed: 1677874297170683785
used memory : 14496 MB
used swap   : 0 MB
seed: 1677874297170683786
used memory : 14822 MB
used swap   : 0 MB
seed: 1677874297170683787
used memory : 15147 MB
used swap   : 0 MB
        SLOW [>1200.000s] risingwave_simulation::it nexmark_chaos::nexmark_chaos_q9_single
seed: 1677874297170683788
used memory : 15433 MB
used swap   : 0 MB
seed: 1677874297170683789
used memory : 15721 MB
used swap   : 0 MB
seed: 1677874297170683790
used memory : 16058 MB
used swap   : 0 MB
seed: 1677874297170683791
used memory : 16386 MB
used swap   : 0 MB
seed: 1677874297170683792
used memory : 16718 MB
used swap   : 0 MB
        SLOW [>1600.000s] risingwave_simulation::it nexmark_chaos::nexmark_chaos_q9_single
seed: 1677874297170683793
used memory : 17027 MB
used swap   : 0 MB
seed: 1677874297170683794
used memory : 17350 MB
used swap   : 0 MB
seed: 1677874297170683795
used memory : 17608 MB
used swap   : 0 MB
seed: 1677874297170683796
used memory : 17962 MB
used swap   : 0 MB
seed: 1677874297170683797
used memory : 18290 MB
used swap   : 0 MB
        SLOW [>2000.000s] risingwave_simulation::it nexmark_chaos::nexmark_chaos_q9_single
seed: 1677874297170683798
used memory : 18577 MB
used swap   : 0 MB
seed: 1677874297170683799
used memory : 18871 MB
used swap   : 0 MB
seed: 1677874297170683800
used memory : 19176 MB
used swap   : 0 MB
seed: 1677874297170683801
used memory : 19519 MB
used swap   : 0 MB
        SLOW [>2400.000s] risingwave_simulation::it nexmark_chaos::nexmark_chaos_q9_single
seed: 1677874297170683802
used memory : 19809 MB
used swap   : 0 MB
seed: 1677874297170683803
used memory : 20166 MB
used swap   : 0 MB
seed: 1677874297170683804
used memory : 20507 MB
used swap   : 0 MB
seed: 1677874297170683805
used memory : 20813 MB
used swap   : 0 MB
seed: 1677874297170683806
used memory : 21128 MB
used swap   : 0 MB
        SLOW [>2800.000s] risingwave_simulation::it nexmark_chaos::nexmark_chaos_q9_single

```

**Comment by xxchan** [comment]
mem profile (via Xcode Instruments on my mac)

st

**Review by wangrunji0408 — APPROVED** [review]
LGTM, thanks!

**Comment by fuyufjh** [comment]
> (nexmark_chaos_q9_single)
> 
> mem profile (via Xcode Instruments on my mac)

Is it due to the operator cache? 🤔

**Pull request merged**